### ISSUE TITLE
@PostConstruct doesn't work for intercepted beans

### ISSUE THREAD

**Issue by l0co** [micronaut-projects/micronaut-core#1564]
In 1.1.0. For this class `@PostConstruct` is never called:

```java
@Transactional @Singleton
public class CategoryService {

	@PostConstruct
	protected void init11() {
		System.out.println("initialing !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
	}

}
```

However, when I remove `@Transactional` it is called properly.

**Comment by l0co** [comment]
A little longer story, while I was debugging what's wrong with this I've stopped at `AbstractBeanDefinition.injectBean()` method. What I can see on the stacktrace is:

```
injectBean:586, AbstractBeanDefinition (io.micronaut.context)
injectBean:-1, $CategoryServiceDefinition (com.twofishsoftware.joshsfrog.api.service.product)
injectBean:-1, $$CategoryServiceDefinition$InterceptedDefinition (com.twofishsoftware.joshsfrog.api.service.product)
build:-1, $$CategoryServiceDefinition$InterceptedDefinition (com.twofishsoftware.joshsfrog.api.service.product)
doCreateBean:1474, DefaultBeanContext (io.micronaut.context)
[...]
```
And then I've looked into classes generated by Micronaut with the decomplier, and here's the result:

```java
class $CategoryServiceDefinition extends AbstractBeanDefinition<CategoryService> implements BeanFactory<CategoryService>, InitializingBeanDefinition<CategoryService> {
    protected $CategoryServiceDefinition(Class var1, AnnotationMetadata var2, boolean var3, Argument[] var4) {
        super(var1, var2, var3, var4);
        super.addPostConstruct(CategoryService.class, "init11", (Argument[])null, new DefaultAnnotationMetadata(AnnotationUtil.internMapOf(new Object[]{"javax.inject.Singleton", Collections.EMPTY_MAP, "io.micronaut.spring.tx.annotation.Transactional", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Executable", Collections.EMPTY_MAP, "io.micronaut.core.annotation.Blocking", Collections.EMPTY_MAP, "javax.inject.Scope", Collections.EMPTY_MAP, "io.micronaut.context.annotation.Type", AnnotationUtil.internMapOf(new Object[]{"value", new AnnotationClassValue[]{$micronaut_load_class_value_0()}}), "io.micronaut.aop.Around", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Executable", Collections.EMPTY_MAP, "io.micronaut.core.annotation.Blocking", Collections.EMPTY_MAP, "javax.inject.Scope", Collections.EMPTY_MAP, "io.micronaut.context.annotation.Type", AnnotationUtil.internMapOf(new Object[]{"value", new AnnotationClassValue[]{$micronaut_load_class_value_0()}}), "io.micronaut.aop.Around", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"javax.inject.Singleton", Collections.EMPTY_MAP, "io.micronaut.spring.tx.annotation.Transactional", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Executable", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional", "io.micronaut.aop.Around"}), "javax.inject.Scope", AnnotationUtil.internListOf(new Object[]{"javax.inject.Singleton"}), "io.micronaut.core.annotation.Blocking", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"}), "io.micronaut.context.annotation.Type", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"}), "io.micronaut.aop.Around", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"})})), false);
    }

    public $CategoryServiceDefinition() {
        this(CategoryService.class, new DefaultAnnotationMetadata((Map)null, (Map)null, AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Executable", Collections.EMPTY_MAP, "io.micronaut.core.annotation.Blocking", Collections.EMPTY_MAP, "javax.inject.Scope", Collections.EMPTY_MAP, "io.micronaut.context.annotation.Type", AnnotationUtil.internMapOf(new Object[]{"value", new AnnotationClassValue[]{$micronaut_load_class_value_0()}}), "io.micronaut.aop.Around", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"javax.inject.Singleton", Collections.EMPTY_MAP, "io.micronaut.spring.tx.annotation.Transactional", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Executable", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional", "io.micronaut.aop.Around"}), "javax.inject.Scope", AnnotationUtil.internListOf(new Object[]{"javax.inject.Singleton"}), "io.micronaut.core.annotation.Blocking", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"}), "io.micronaut.context.annotation.Type", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"}), "io.micronaut.aop.Around", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"})})), false, (Argument[])null);
    }

    public CategoryService build(BeanResolutionContext var1, BeanContext var2, BeanDefinition<CategoryService> var3) {
        CategoryService var4 = new CategoryService();
        var4 = (CategoryService)this.injectBean(var1, var2, var4);
        CategoryService var10000 = var4;
        var4 = (CategoryService)this.initialize(var1, var2, var4);
        return var10000;
    }

    protected Object injectBean(BeanResolutionContext var1, BeanContext var2, Object var3) {
        CategoryService var4 = (CategoryService)var3;
        Object var5 = null;
        return super.injectBean(var1, (DefaultBeanContext)var2, var3);
    }

    public CategoryService initialize(BeanResolutionContext var1, BeanContext var2, CategoryService var3) {
        CategoryService var4 = (CategoryService)var3;
        super.postConstruct(var1, (DefaultBeanContext)var2, var3);
        var4.init11();
        return var4;
    }

    protected AnnotationMetadata resolveAnnotationMetadata() {
        return CategoryServiceDefinitionClass.$ANNOTATION_METADATA;
    }
}

class $$CategoryServiceDefinition$InterceptedDefinition extends CategoryServiceDefinition implements BeanFactory<Intercepted>, ProxyBeanDefinition<Intercepted> {
    protected $$CategoryServiceDefinition$InterceptedDefinition(Class var1, AnnotationMetadata var2, boolean var3, Argument[] var4) {
        super(var1, var2, var3, var4);
    }

    public $$CategoryServiceDefinition$InterceptedDefinition() {
        this(Intercepted.class, new DefaultAnnotationMetadata((Map)null, (Map)null, AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Executable", Collections.EMPTY_MAP, "io.micronaut.core.annotation.Blocking", Collections.EMPTY_MAP, "javax.inject.Scope", Collections.EMPTY_MAP, "io.micronaut.context.annotation.Type", AnnotationUtil.internMapOf(new Object[]{"value", new AnnotationClassValue[]{$micronaut_load_class_value_0()}}), "io.micronaut.aop.Around", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"javax.inject.Singleton", Collections.EMPTY_MAP, "io.micronaut.spring.tx.annotation.Transactional", Collections.EMPTY_MAP}), AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Executable", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional", "io.micronaut.aop.Around"}), "javax.inject.Scope", AnnotationUtil.internListOf(new Object[]{"javax.inject.Singleton"}), "io.micronaut.core.annotation.Blocking", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"}), "io.micronaut.context.annotation.Type", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"}), "io.micronaut.aop.Around", AnnotationUtil.internListOf(new Object[]{"io.micronaut.spring.tx.annotation.Transactional"})})), false, new Argument[]{Argument.of(BeanContext.class, "beanContext", (AnnotationMetadata)null, (Argument[])null), Argument.of(Interceptor[].class, "interceptors", new DefaultAnnotationMetadata(AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Type", AnnotationUtil.internMapOf(new Object[]{"value", new Object[]{$micronaut_load_class_value_1()}})}), (Map)null, (Map)null, AnnotationUtil.internMapOf(new Object[]{"io.micronaut.context.annotation.Type", AnnotationUtil.internMapOf(new Object[]{"value", new Object[]{$micronaut_load_class_value_1()}})}), (Map)null), (Argument[])null)});
    }

    public Intercepted build(BeanResolutionContext var1, BeanContext var2, BeanDefinition<Intercepted> var3) {
        Intercepted var4 = new Intercepted((BeanContext)super.getBeanForConstructorArgument(var1, var2, 0), (Interceptor[])super.getBeanForConstructorArgument(var1, var2, 1));
        var4 = (Intercepted)this.injectBean(var1, var2, var4);
        return var4;
    }

    protected Object injectBean(BeanResolutionContext var1, BeanContext var2, Object var3) {
        Intercepted var4 = (Intercepted)var3;
        Object var5 = null;
        return super.injectBean(var1, (DefaultBeanContext)var2, var3);
    }

    public Class getTargetDefinitionType() {
        return CategoryServiceDefinition.class;
    }

    public Class getTargetType() {
        return CategoryService.class;
    }

    protected AnnotationMetadata resolveAnnotationMetadata() {
        return InterceptedDefinitionClass.$ANNOTATION_METADATA;
    }
}
```

What I can see, if the bean is not intercepted and the invocation goes through `$CategoryServiceDefinition.build()` the `initialize()` method with post constructs is executed properly. But when I have an intercepted bean and things goes through `$$CategoryServiceDefinition$InterceptedDefinition.build()` no `initialize()` is called and we have no `@PostConstruct` method working.

**Comment by l0co** [comment]
Just checked for curiosity, such a strange thing like this:

```java
@Transactional @Factory
public class CategoryService {

	@Bean
	public Object bean() {
		return new Object();
	}

}
```

`bean()` method is not invoked as well.

**Comment by l0co** [comment]
Any option to get this fix from gradle before official 1.1.1 release?
